File: Python/array/toeplitz_matrix.py
from typing import List

def isToeplitzMatrix(matrix: List[List[int]]) -> bool:
    row = len(matrix)
    col = len(matrix[0])
    for i in range(row):
        r, c = i, 0
        while r + 1 < row and c + 1 < col:
            if matrix[r][c] != matrix[r + 1][c + 1]:
                return False
            r += 1
            c += 1
    for i in range(1, col):
        r, c = 0, i
        while r + 1 < row and c + 1 < col:
            if matrix[r][c] != matrix[r + 1][c + 1]:
                return False
            r += 1
            c += 1
    return True

if __name__ == "__main__":
    print(isToeplitzMatrix([[1,2],[2,2]]))

File: src/tmp/Java/src/com/dbc/algorithm/Array/ToeplitzMatrix.java
package com.dbc.algorithm.Array;

public class ToeplitzMatrix {
    public static boolean isToeplitzMatrix(int[][] matrix) {
        int row = matrix.length, col = matrix[0].length;
        for (int i = 0; i < row - 1; i++) {
            for (int j = 0; j < col - 1; j++) {
                if (matrix[i][j] != matrix[i + 1][j + 1])
                    return false;
            }
        }
        return true;
    }
}
